File: 백준/Silver/10828. 스택/스택.py
import sys

input = sys.stdin.readline

N = int(input())
A = [input().rstrip().split() for _ in range(N)]
stack = []

for a in A :
    if a[0] == "push" :
        stack.append(a[1])
    if a[0] == "pop" :
        if len(stack) > 0 :
            print(stack.pop())
        else :
            print(-1)
    if a[0] == "size" :
        print(len(stack))
    if a[0] == "empty" :
        if len(stack) > 0 :
            print(0)
        else :
            print(1)
    if a[0] == "top" :
        if len(stack) > 0 :
            print(stack[-1])
        else :
            print(-1)

File: 백준/Silver/10828. 스택/스택.js
const readLine = require('fs').readFileSync('dev/stdin').toString().trim();
const [N, ...input] = readLine.split('\n');
const stack = [];
const answer = [];
for (const x of input) {
  const [command, num] = x.split(' ');

  if (command === 'push') {
    stack.push(num);
  } else if (command === 'pop') {
    if (stack.length === 0) {
      answer.push(-1);
    } else {
      answer.push(stack.pop());
    }
  } else if (command === 'size') {
    answer.push(stack.length);
  } else if (command === 'empty') {
    answer.push(stack.length === 0 ? 1 : 0);
  } else if (command === 'top') {
    answer.push(stack.length === 0 ? -1 : stack[stack.length - 1]);
  }
}
console.log(answer.join('\n'));
